File: 0455-assign-cookies/0455-assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        g.sort()
        s.sort()

        count, i, j = 0, 0, 0
        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                count += 1
                i += 1
                j += 1
                continue
            j += 1

        return count
        

File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int count = 0, i = 0, j = 0;
        while(i < g.size() && j < s.size()) {
            if(g[i] <= s[j]) {
                count++;
                i++, j++;
                continue;
            }
            j++; 
        }

        return count;
    }
};